contracts/Pod.sol

```diff
@@ -68,79 +81,121 @@ contract Pod is Initializable, ReentrancyGuardUpgradeSafe, ERC777UpgradeSafe, Ba
 
-        // Set Timestamp for Caller
-        unlockTimestamp[_sender] = _timestamp;
+        // Log event
+        emit PodRedeemedWithTimelock(_sender, _from, _timestamp, _assets, _shares, _tickets);
+        return _assets;
+    }
 
-        // Burn Pod-Share tokens
-        _burn(_sender, _shares);
+    function mintSponsorship(address _receiver, uint256 _amount) external nonReentrant {
+        // Buy Tickets for caller
+        address _sender = _msgSender();
+        _buyTickets(_sender, _amount);
+
+        // Mint Sponsorship Tokens equal to Amount of Collateral supplied
+        podSponsorship.mint(_receiver, _amount);
 
         // Log event
-        emit PodRedeemedWithTimelock(_sender, _timestamp, _assetsRedeemed, _shares, _tickets);
+        emit PodSponsored(_sender, _receiver, _amount);
+    }
+
+    function redeemSponsorshipInstantly(address _from, uint256 _sponsorshipTokens) external nonReentrant returns (uint256) {
+        address _sender = _msgSender();
+        require(_sender == _from || isOperatorFor(_sender, _from), "Pod: Invalid operator");
+        uint256 _balance = podSponsorship.balanceOf(_from);
+        require(_balance >= _sponsorshipTokens, "Pod: Insufficient sponsorship balance");
+
+        // Redeem Sponsorship Tokens for Assets (less the Fairness-Fee)
+        uint256 _assets = ticket().redeemTicketsInstantly(_sponsorshipTokens);
+
+        // Burn the Sponsorship Tokens
+        podSponsorship.burn(_from, _sponsorshipTokens);
 
-        // Return amount of unlocked assets redeemed during sweep
-        return _assetsRedeemed;
+        // Transfer Redeemed Assets to Caller
+        yieldService().token().transfer(_from, _assets);
+
+        // Log event
+        emit PodSponsorRedeemed(_sender, _from, _sponsorshipTokens, _assets);
+        return _assets;
     }
 
+    function redeemSponsorshipWithTimelock(address _from, uint256 _sponsorshipTokens) external nonReentrant returns (uint256) {
+        address _sender = _msgSender();
+        require(_sender == _from || isOperatorFor(_sender, _from), "Pod: Invalid operator");
+        uint256 _balance = podSponsorship.balanceOf(_from);
+        require(_balance >= _sponsorshipTokens, "Pod: Insufficient sponsorship balance");
+
+        // Sweep Previously Unlocked Assets for Caller
+        uint256 _assets = sweepForUser(_from);
+
+        // Redeem Sponsorship Tokens with Timelock
+        uint256 _timestamp = _redeemTicketsWithTimelock(_from, _sponsorshipTokens);
+
+        // Burn the Sponsorship Tokens
+        podSponsorship.burn(_from, _sponsorshipTokens);
+
+        // Transfer any funds that are already unlocked
+        if (_timestamp <= block.timestamp) {
+          _assets = _assets.add(sweepForUser(_from));
+        }
+
+        // Log event
+        emit PodSponsorRedeemedWithTimelock(_sender, _from, _timestamp, _sponsorshipTokens, _assets);
+        return _assets;
+    }
 
     function sweepForUser(address _user) public returns (uint256) {
```

Comment:
Ah- missed this.

We want to call `sweepForUser` for many users at once, since it's an expensive operation.  Let's change this to an array:

`sweepForUser([address])`

Then we can efficiently sweep for multiple users at once